capstone/backend/src/main/java/com/google/servlets/ProductDataServlet.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.java.spanner;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+ @WebServlet("/get-products")
+ public final class ProductDataServlet extends HttpServlet {
+
+  private class PageRequest {
+    int page;
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    Gson g = new Gson();
+    String jsonString = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
+    PageRequest p = g.fromJson(jsonString, PageRequest.class);
+    if (p.page < 0) {
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid request syntax.");
+      return;
+    }
+    List<String> items = LibraryFunctions.getItemTypes(p.page);
```

Comment:
since your `LibraryFunctions.getItemTypes()` implemented [here](https://github.com/googleinterns/step39-2020/blob/15f43e255380cc5c91d8c641cdd29c6e45d06bda/capstone/backend/src/main/java/com/google/spanner/LibraryFunctions.java#L114) throws a SpannerException, should put this in a try/catch block so that we can catch the exception.